### PR TITLE
updated the code to use the findOne instead of dbref

### PR DESCRIPTION
**Pull request by dtracers**

### CODE REVIEW

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java [hunk 1]
```diff
@@ -464,11 +464,10 @@ public static boolean mongoUpdateAssignment(final Authenticator authenticator, f
...
     static boolean mongoInsert(final DB dbs, final String assignmentId, final String problemId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, ASSIGNMENT_COLLECTION, convertStringToObjectId(assignmentId));
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
+        final DBObject cursor = collection.findOne(convertStringToObjectId(assignmentId));;
 
-        final DBObject cursor = myDbRef.fetch();
-
-        final DBCollection courses = dbs.getCollection(ASSIGNMENT_COLLECTION);
+        final DBCollection courses = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: When is this being used and isn't this just a copy of the collection variable?

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java [hunk 2]
```diff
@@ -491,9 +490,9 @@ static boolean mongoInsert(final DB dbs, final String assignmentId, final String
      */
     // package-private
     static void mongoInsertDefaultGroupId(final DB dbs, final String assignmentId, final List<String>... ids) {
-        final DBRef myDbRef = new DBRef(dbs, ASSIGNMENT_COLLECTION, new ObjectId(assignmentId));
+        final DBRef myDbRef = new DBRef(dbs, getCollectionFromType(School.ItemType.ASSIGNMENT), new ObjectId(assignmentId));
         final DBObject corsor = myDbRef.fetch();
-        final DBCollection assignments = dbs.getCollection(ASSIGNMENT_COLLECTION);
+        final DBCollection assignments = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: Why is this object called collections in other methods??

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java [hunk 3]
```diff
@@ -89,7 +89,7 @@ private AssignmentManager() {
      */
     public static String mongoInsertAssignment(final Authenticator authenticator, final DB dbs, final String authId, final SrlAssignment assignment)
             throws AuthenticationException, DatabaseAccessException {
-        final DBCollection assignmentCollection = dbs.getCollection(ASSIGNMENT_COLLECTION);
+        final DBCollection assignmentCollection = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: This variable has 3 names depending on which method it is used in: collection, assignments, and assignmentCollection. Can you just pick one of the 3 and standardize it?

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java [hunk 4]
```diff
@@ -157,8 +157,8 @@ public static String mongoInsertAssignment(final Authenticator authenticator, fi
     @SuppressWarnings({ "PMD.CyclomaticComplexity", "PMD.ModifiedCyclomaticComplexity", "PMD.StdCyclomaticComplexity" })
     public static SrlAssignment mongoGetAssignment(final Authenticator authenticator, final DB dbs, final String authId, final String assignmentId,
             final long checkTime) throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, ASSIGNMENT_COLLECTION, convertStringToObjectId(assignmentId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: Change  this to assignmentCollection or assignments please cause collection also means:
`database.getCollection(USER_GROUP_COLLECTION);`

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java [hunk 5]
```diff
@@ -346,14 +346,14 @@ private static void setAssignmentStateAndDate(final SrlAssignment.Builder exactA
...
+        final DBObject cursor = collection.findOne(convertStringToObjectId(assignmentId));
 
         if (cursor == null) {
             throw new DatabaseAccessException("Assignment was not found with the following ID: " + assignmentId, true);
         }
 
-        final DBCollection assignmentCollection = dbs.getCollection(ASSIGNMENT_COLLECTION);
+        final DBCollection assignmentCollection = dbs.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: Isn't this identical to collection in line 349?

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java [hunk 6]
```diff
@@ -122,8 +121,8 @@ public static String mongoInsertBankProblem(final DB dbs, final SrlBankProblem p
      */
     public static SrlBankProblem mongoGetBankProblem(final Authenticator authenticator, final DB dbs, final String authId, final String problemBankId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, PROBLEM_BANK_COLLECTION, convertStringToObjectId(problemBankId));
-        final DBObject mongoBankProblem = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java]: please change the name to problemBank or problemBankCollection

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java [hunk 7]
```diff
@@ -215,8 +214,8 @@ private static SrlBankProblem extractBankProblem(final DBObject dbObject, final
     public static boolean mongoUpdateBankProblem(final Authenticator authenticator, final DB dbs, final String authId, final String problemBankId,
             final SrlBankProblem problem) throws AuthenticationException, DatabaseAccessException {
         boolean update = false;
-        final DBRef myDbRef = new DBRef(dbs, PROBLEM_BANK_COLLECTION, convertStringToObjectId(problemBankId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java]: please change the name to problemBank or problemBankCollection

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java [hunk 8]
```diff
@@ -361,8 +360,8 @@ public static boolean mongoUpdateBankProblem(final Authenticator authenticator,
     public static String mongoGetRegistrationKey(final Authenticator authenticator, final DB database,
             final String authId, final String bankProblemId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(database, PROBLEM_BANK_COLLECTION, convertStringToObjectId(bankProblemId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = database.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java]: please change the name to problemBank or problemBankCollection or problemCollwction

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 9]
```diff
@@ -121,8 +121,9 @@ static String mongoInsertCourse(final DB dbs, final SrlCourse course) {
     static SrlCourse mongoGetCourse(final Authenticator authenticator, final DB dbs, final String authId, final String courseId,
             final long checkTime)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, COURSE_COLLECTION, convertStringToObjectId(courseId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: change name here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 10]
```diff
@@ -221,15 +222,15 @@ static SrlCourse mongoGetCourse(final Authenticator authenticator, final DB dbs,
     static boolean mongoUpdateCourse(final Authenticator authenticator, final DB dbs, final String authId, final String courseId,
             final SrlCourse course) throws AuthenticationException, DatabaseAccessException {
         boolean update = false;
-        final DBRef myDbRef = new DBRef(dbs, COURSE_COLLECTION, convertStringToObjectId(courseId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 11]
```diff
@@ -319,14 +320,14 @@ static boolean mongoUpdateCourse(final Authenticator authenticator, final DB dbs
      */
     static boolean mongoInsertAssignmentIntoCourse(final DB dbs, final String courseId, final String assignmentId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, COURSE_COLLECTION, convertStringToObjectId(courseId));
-        final DBObject corsor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 12]
```diff
@@ -351,10 +352,10 @@ static boolean mongoInsertAssignmentIntoCourse(final DB dbs, final String course
      */
     static boolean mongoInsertLectureIntoCourse(final DB dbs, final String courseId, final String lectureId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, COURSE_COLLECTION, convertStringToObjectId(courseId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 13]
```diff
@@ -351,10 +352,10 @@ static boolean mongoInsertAssignmentIntoCourse(final DB dbs, final String course
...
+        final DBObject cursor = collection.findOne(convertStringToObjectId(courseId));
         DBObject updateObj = null;
-        final DBCollection courses = dbs.getCollection(COURSE_COLLECTION);
+        final DBCollection courses = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: when is this object used and why is it a copy of collection? why not use collection?

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java [hunk 14]
```diff
@@ -503,8 +504,8 @@ static void mongoInsertDefaultGroupId(final DB dbs, final String courseId, final
     public static String mongoGetRegistrationKey(final Authenticator authenticator, final DB database, final String authId, final String courseId,
             final boolean checkTeacher)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(database, COURSE_COLLECTION, convertStringToObjectId(courseId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = database.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java [hunk 15]
```diff
@@ -119,8 +119,9 @@ public static String mongoInsertCourseProblem(final Authenticator authenticator,
      */
     public static SrlProblem mongoGetCourseProblem(final Authenticator authenticator, final DB dbs, final String authId, final String problemId,
             final long checkTime) throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, COURSE_PROBLEM_COLLECTION, convertStringToObjectId(problemId));
-        final DBObject cursor = myDbRef.fetch();
+
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java [hunk 16]
```diff
@@ -223,15 +224,15 @@ public static SrlProblem mongoGetCourseProblem(final Authenticator authenticator
     public static boolean mongoUpdateCourseProblem(final Authenticator authenticator, final DB dbs, final String authId, final String problemId,
             final SrlProblem problem) throws AuthenticationException, DatabaseAccessException {
         boolean update = false;
-        final DBRef myDbRef = new DBRef(dbs, COURSE_PROBLEM_COLLECTION, convertStringToObjectId(problemId));
-        final DBObject cursor = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java [hunk 17]
```diff
@@ -223,15 +224,15 @@ public static SrlProblem mongoGetCourseProblem(final Authenticator authenticator
...
+        final DBObject cursor = collection.findOne(convertStringToObjectId(problemId));
 
         if (cursor == null) {
             throw new DatabaseAccessException("Course problem was not found with the following ID: " + problemId);
         }
 
         final BasicDBObject updateObj = new BasicDBObject();
-        final DBCollection problemCollection = dbs.getCollection(COURSE_PROBLEM_COLLECTION);
+        final DBCollection problemCollection = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java [hunk 18]
```diff
@@ -277,9 +278,9 @@ public static boolean mongoUpdateCourseProblem(final Authenticator authenticator
      *         The list of id groupings that contain the ids being copied over.
      */
     static void mongoInsertDefaultGroupId(final DB dbs, final String courseProblemId, final List<String>... ids) {
-        final DBRef myDbRef = new DBRef(dbs, COURSE_PROBLEM_COLLECTION, new ObjectId(courseProblemId));
+        final DBRef myDbRef = new DBRef(dbs, getCollectionFromType(School.ItemType.COURSE_PROBLEM), new ObjectId(courseProblemId));
         final DBObject corsor = myDbRef.fetch();
-        final DBCollection problems = dbs.getCollection(COURSE_PROBLEM_COLLECTION);
+        final DBCollection problems = dbs.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/CourseProblemManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/GradingPolicyManager.java [hunk 19]
```diff
@@ -172,8 +174,9 @@ public static void insertGradingPolicy(final Authenticator authenticator, final
      */
     static ProtoGradingPolicy getGradingPolicy(final Authenticator authenticator, final DB dbs, final String courseId, final String userId)
             throws AuthenticationException, DatabaseAccessException {
-        final DBRef myDbRef = new DBRef(dbs, GRADING_POLICY_COLLECTION, new ObjectId(courseId));
-        final DBObject policyObject = myDbRef.fetch();
+        final DBCollection collection = dbs.getCollection(GRADING_POLICY_COLLECTION);
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/GradingPolicyManager.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/AssignmentManagerTest.java [hunk 20]
```diff
@@ -143,8 +145,8 @@ public void insertAssignmentIntoAssignmentSetsValuesCorrectly() throws Exception
 
         assignmentId = AssignmentManager.mongoInsertAssignment(authenticator, db, ADMIN_USER, defaultAssignment.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.ASSIGNMENT, true), new ObjectId(assignmentId));
-        final DBObject mongoAssignment = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/AssignmentManagerTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java [hunk 21]
```diff
@@ -102,8 +104,9 @@ public void insertBankProblemNoPermissions() throws Exception {
 
         String problemBankId = BankProblemManager.mongoInsertBankProblem(db, bankProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, PROBLEM_BANK_COLLECTION, new ObjectId(problemBankId));
-        final DBObject mongoBankProblem = myDbRef.fetch();
+
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java [hunk 22]
```diff
@@ -122,9 +125,10 @@ public void insertBankProblemPermissions() throws Exception {
 
         String problemBankId = BankProblemManager.mongoInsertBankProblem(db, bankProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, PROBLEM_BANK_COLLECTION, new ObjectId(problemBankId));
-        final DBObject mongoBankProblem = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/CourseManagerTest.java [hunk 23]
```diff
@@ -113,8 +116,8 @@ public void insertCourseIntoCourseWithValidPermissionAndAllValuesAreSetCorrectly
         defaultCourse.setName(VALID_NAME);
         String courseId = CourseManager.mongoInsertCourse(db, defaultCourse.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.COURSE, true), new ObjectId(courseId));
-        final DBObject mongoCourse = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/CourseManagerTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/CourseProblemManagerTest.java [hunk 24]
```diff
@@ -164,8 +167,8 @@ public void insertCourseProblemIntoAssignmentWithValidPermission() throws Except
 
         courseProblemId = CourseProblemManager.mongoInsertCourseProblem(authenticator, db, ADMIN_USER, defaultProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.COURSE_PROBLEM, true), new ObjectId(courseProblemId));
-        final DBObject mongoProblem = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/CourseProblemManagerTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java [hunk 25]
```diff
@@ -209,8 +211,8 @@ public void insertingBankProblemCreatesRegistrationKey() throws AuthenticationEx
 
         String problemBankId = institution.insertBankProblem(TEACHER_USER_ID, TEACHER_AUTH_ID, bankProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, PROBLEM_BANK_COLLECTION, new ObjectId(problemBankId));
-        final DBObject mongoBankProblem = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java [hunk 26]
```diff
@@ -239,8 +241,8 @@ public void insertingCourseCreatesRegistrationKey() throws AuthenticationExcepti
 
         String courseId = institution.insertCourse(TEACHER_USER_ID, TEACHER_AUTH_ID, defaultCourse.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.COURSE, true), new ObjectId(courseId));
-        final DBObject mongoCourse = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java [hunk 27]
```diff
@@ -272,8 +274,8 @@ public void insertingAssignmentAlsoCreatesUserGroup() throws Exception {
 
         assignmentId = institution.insertAssignment(TEACHER_USER_ID, TEACHER_AUTH_ID, defaultAssignment.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.ASSIGNMENT, true), new ObjectId(assignmentId));
-        final DBObject mongoAssignment = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.ASSIGNMENT));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java [hunk 28]
```diff
@@ -300,8 +302,8 @@ public void insertCourseProblemCreatesUserGroup() throws Exception {
 
         courseProblemId = institution.insertCourseProblem(TEACHER_USER_ID, TEACHER_AUTH_ID, defaultProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, DbSchoolUtility.getCollectionFromType(School.ItemType.COURSE_PROBLEM, true), new ObjectId(courseProblemId));
-        final DBObject mongoProblem = myDbRef.fetch();
+        final DBCollection collection = db.getCollection(getCollectionFromType(School.ItemType.COURSE_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java [hunk 29]
```diff
@@ -328,8 +330,8 @@ public void insertCourseProblemInsertsBankProblemPermissions() throws Exception
...
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/MongoInstitutionTest.java]: and here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/utilities/BreakDatabase.java [hunk 30]
```diff
@@ -66,7 +67,7 @@ public BreakDatabase(final DB db) {
         String courseID = mongoDatabase.insertCourse(null, user.getUsername(), course);
         mongoDatabase.putUserInCourse(null, user.getUsername(), courseID, null);
 
-        DBCollection collection = database.getCollection(COURSE_COLLECTION);
+        DBCollection collection = database.getCollection(getCollectionFromType(School.ItemType.COURSE));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/utilities/BreakDatabase.java]: here

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/utilities/BreakDatabase.java [hunk 31]
```diff
@@ -80,7 +81,7 @@ public BreakDatabase(final DB db) {
...
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/utilities/BreakDatabase.java]: here

**Comment by areftw** [comment]
Please standardize the names of the numerous types of Collection Objects

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java (hunk 3 above)
Comment (dtracers) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/AssignmentManager.java]: picking assignmentCollection as the standard

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java [hunk 32]
```diff
@@ -227,7 +226,7 @@ public static boolean mongoUpdateBankProblem(final Authenticator authenticator,
                 .build();
         final AuthenticationResponder responder = authenticator
                 .checkAuthentication(School.ItemType.BANK_PROBLEM, problemBankId, authId, 0, authType);
-        final DBCollection problemCollection = dbs.getCollection(PROBLEM_BANK_COLLECTION);
+        final DBCollection problemCollection = dbs.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java]: backProblemCollection here plox

File: CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java [hunk 33]
```diff
@@ -329,7 +328,7 @@ public static boolean mongoUpdateBankProblem(final Authenticator authenticator,
             throw new AuthenticationException(AuthenticationException.INVALID_PERMISSION);
         }
 
-        final DBCollection problemCollection = database.getCollection(PROBLEM_BANK_COLLECTION);
+        final DBCollection problemCollection = database.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/BankProblemManager.java]: bankProblemCollection here too

File: CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java [hunk 34]
```diff
@@ -122,9 +125,10 @@ public void insertBankProblemPermissions() throws Exception {
 
         String problemBankId = BankProblemManager.mongoInsertBankProblem(db, bankProblem.build());
 
-        final DBRef myDbRef = new DBRef(db, PROBLEM_BANK_COLLECTION, new ObjectId(problemBankId));
-        final DBObject mongoBankProblem = myDbRef.fetch();
+        final DBCollection bankProblemCollection = db.getCollection(getCollectionFromType(School.ItemType.BANK_PROBLEM));
+        final DBObject cursor = bankProblemCollection.findOne(convertStringToObjectId(problemBankId));
 
+        Assert.assertNotNull(cursor);
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchDatabaseServer/src/test/java/database/institution/mongo/BankProblemManagerTest.java]: Aren't Assert's mainly used for programmers while coding and should be removed when deploying?

**Pull request merged**